core/src/main/java/com/microsoft/applicationinsights/internal/agent/CoreAgentNotificationsHandler.java

```diff
@@ -170,7 +170,7 @@ public void httpMethodFinishedWithPath(String identifier, String method, String
         }
         long deltaInMS = nanoToMilliseconds(delta);
         String name = method + " " + path;
-        RemoteDependencyTelemetry telemetry = new RemoteDependencyTelemetry(name, uri, new Duration(delta), true);
+        RemoteDependencyTelemetry telemetry = new RemoteDependencyTelemetry(name, uri, new Duration(deltaInMS), true);
```

Comment:
repeating the comment, it was collapsed by the commit
Well, there is no code that sets telemetry.timestamp in this method. If it's not set here, it will be set in TelemetryClient.track to Now

        if (telemetry.getTimestamp() == null) {
            telemetry.setTimestamp(new Date());
        }
i.e. duration is correct, but start time is not. I have not tried it though, it would be great if you can confirm it's not the case.

Reply:
@lmolkova yes it's correct that there is no place where this is set explicitly and that means that TelemetryClient will set it to now which is actually after the completion of the dependency execution. The comment above the method though says that "time at which the telemetry is recorded" - I thought this means when the telemetry object is created because I do not see this any where being reported apart from that of RequestTelemetry object. Thoughts?

Reply:
If we really want to record the start time, one thing that can be done is that when we create the DependencyTelemetry basically subtract the duration from the timeNow. Still it would have some nano seconds of error but that is the solution I have in my mind. Though it would not be related to this PR then. There are multiple places to fix that in whole CoreAgentNotificationHandler class. If you have better solution let me know.

Reply:
timestamp is definitely a  moment when dependency call was started, not when dependency telemetry was created (i.e. call finished).

I'm fine if you just subtract delta from Now,
It could also be done by storing start time in httpRequestStart method, but it's much more complicated.

we had this problem before in .NET. Here is how it looks like: https://github.com/Microsoft/ApplicationInsights-dotnet-server/pull/670,

It's confusing for users and breaks our portal experience: dependency starts at the end of request and finishes after request is completed. It's hard to understand what causes request to took so much time, etc.

So it would be great if you made sure start time is set correctly in this PR.

What are other places? You mean other kinds of dependencies? Let's fix http since we are here already and it's most important and create an issue to revisit other dependency tracking calls

Reply:
sure. yeah, instrumenting it would be a difficult to maintain thing. I would love to instrument as less as possible unless it warrants the criticalness. I would subtract it for now as best alternative and relatively less loss of precision. 

And yes, I was referring to other dependencies.

Reply:
This is fixed. Attached pictures from UI indicating the same.

Before fix:
![image](https://user-images.githubusercontent.com/16353121/36513938-e66b01e4-1726-11e8-8544-7073803fdda6.png)


After Fix:
![image](https://user-images.githubusercontent.com/16353121/36513882-a3138178-1726-11e8-9867-cde13be8b82c.png)